feat(ui)!: provide magnifier configuration parameter to form builder text field

tsinis opened the pull request:
## Connection with issue(s)

Connected to Flutter SDK bug: https://github.com/flutter/flutter/issues/119711

## Solution description

In version 3.7 Flutter team [introduced](https://medium.com/flutter/whats-new-in-flutter-3-7-38cbea71133c) **Text magnifier** feature, however, did not provide a way how to disable it globally. But this feature has [bugs](https://github.com/flutter/flutter/issues/119711) related to scrollable widgets. As a workaround, there is an option to provide [magnifierConfiguration](https://github.com/flutter/flutter/pull/107477) to `TextField`/`EditableText` widgets. Since `FormBuilderTextField` is using `TextField` under the hood there was no way how to provide this configuration to `FormBuilderTextField`, this PR fixes this issue.

## To Do

- [x] Read [contributing guide](https://github.com/flutter-form-builder-ecosystem/.github/blob/main/CONTRIBUTING.md)
- [x] Check the original issue to confirm it is fully satisfied
- [x] Add solution description to help guide reviewers
- [x] If apply, add documentation to code properties and package readme

tsinis commented:
Failing on "info"s, at analyze part, however, contributing guide mentioning [warnings/errors only](https://github.com/flutter-form-builder-ecosystem/.github/blob/main/CONTRIBUTING.md#make-a-pull-request) :-/. I will try if it's possible to fix it without any big harm or breaking changes tomorrow.

deandreamatias commented:
Hi @tsinis
Thanks for contribution. I changed the contribution file, but witch you said is true, the document only refer to warning and errors. But, if you can fix this info lints, will be a nice contribution (one more) to package. I will really appreciate that

tsinis commented:
> Hi @tsinis Thanks for contribution. I changed the contribution file, but witch you said is true, the document only refer to warning and errors. But, if you can fix this info lints, will be a nice contribution (one more) to package. I will really appreciate that

Hey, @deandreamatias I've fixed the infos (replaced the deprecated field with a new one). By the way, you should add the `pubspec.lock` file to the .gitignore, this is not a file that should be committed to the package: https://dart.dev/guides/libraries/private-files#pubspeclock
